flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/PythonCalcSplitRule.scala

```diff
@@ -168,12 +168,34 @@ abstract class PythonCalcSplitProjectionRuleBase(description: String)
   }
 }
 
+abstract class PythonCalcSplitRexFieldRuleBase(description: String)
+  extends PythonCalcSplitRuleBase(description) {
+
+  override def needConvert(program: RexProgram, node: RexNode): Boolean = {
+    node match {
+      case x: RexFieldAccess => x.getReferenceExpr match {
+        case y: RexLocalRef if isPythonCall(program.expandLocalRef(y)) => true
+        case _ => false
+      }
+      case _ => false
+    }
+  }
+
+  protected def containsFieldAccessAfterPythonCall(node: RexNode): Boolean = {
+    node match {
+      case call: RexCall => call.getOperands.exists(containsFieldAccessAfterPythonCall)
+      case x: RexFieldAccess => isPythonCall(x.getReferenceExpr)
```

Comment:
We should call the `containsFieldAccessAfterPythonCall` here if the expression is not a python call to handle the situation like `where javaRowFunc(pyRowFunc5(a)).f0 is NULL`. And maybe we can combine this logic into `FunctionFinder` as the field access, java call, python call may appear alternately. That should be able to better to cover various situations.

Reply:
Yes, current logic can't deal with the case of `where javaRowFunc(pyRowFunc5(a)).f0 is NULL`. We need to change `isPythonCall ` to `containsPythonCall` and add the logic of `visitFieldAccess` into the `FunctionFinder` of `PythonUtil`